02-peer-protocol.md

```diff
@@ -409,25 +409,61 @@ this channel will continue to use `option_static_remotekey` or `option_anchor_ou
 
 This message indicates that the funding transaction has reached the `minimum_depth` asked for in `accept_channel`. Once both nodes have sent this, the channel enters normal operating mode.
 
+Nodes which have funded the channel, or trust their peers to have
+done, or wish to use `push_msat` from other side's opening (without
+accepting incoming HTLCs) can simply start using the channel instantly
+by sending `funding_locked`.
+
+
 1. type: 36 (`funding_locked`)
 2. data:
     * [`channel_id`:`channel_id`]
     * [`point`:`next_per_commitment_point`]
+    * [`funding_locked_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `funding_locked_tlvs`
+2. types:
+    1. type: 1 (`short_channel_id`)
+    2. data:
+        * [`short_channel_id`:`alias`]
 
 #### Requirements
 
-The sender MUST:
-  - NOT send `funding_locked` unless outpoint of given by `funding_txid` and
+The sender:
+  - MUST NOT send `funding_locked` unless outpoint of given by `funding_txid` and
    `funding_output_index` in the `funding_created` message pays exactly `funding_satoshis` to the scriptpubkey specified in [BOLT #3](03-transactions.md#funding-transaction-output).
-  - wait until the funding transaction has reached `minimum_depth` before
-  sending this message.
-  - set `next_per_commitment_point` to the per-commitment point to be used
-  for the following commitment transaction, derived as specified in
+  - MUST set `next_per_commitment_point` to the per-commitment point to be used
+  for commitment transaction #1, derived as specified in
   [BOLT #3](03-transactions.md#per-commitment-secret-requirements).
+  - if `option_scid_alias_only` was negotiated:
+    - MUST set `short_channel_id` `alias`.
+  - otherwise:
+    - SHOULD set `short_channel_id` `alias`.
+  - if it sets `alias`:
+    - if the `announce_channel` bit was set in both `open_channel` and `accept_channel`:
+      - SHOULD initially set `alias` to value not related to the real `short_channel_id`.
+      - SHOULD re-transmit `funding_locked` with the real `short_channel_id` once the channel is confirmed.
+    - otherwise:
+      - MUST set `alias` to a value not related to the real `short_channel_id`.
+    - MUST NOT send the same `alias` for multiple peers.
+    - MUST always recognize the `alias` as a `short_channel_id` for incoming HTLCs to this channel.
+    - if `option_scid_alias_only` was negotiated:
+      - MUST NOT allow incoming HTLCs to this channel using the real `short_channel_id`
```

Comment:
If `option_scid_alias_only` is negotiated and incoming HTLCs of the real `short_channel_id`, does this break the spontaneous nature of key-send? The payer will need to know one of the alias `short_channel_id` which will be different from the publicly announced "real" one in the `channel_announcement` / `channel_update` messages.  Additionally, regular routing may require a payer to know aliases of ALL `option_scid_alias_only`-negotiating hops along the way.

Reply:
It also makes sense in the zero-conf case before the scid is known

Reply:
Shouldn't this include private zero-conf channels?

Reply:
Its already implied cause you're not allowed to set a public channel as `scid_alias`.

Reply:
The clause you're referencing only requires that if the option_scid_alias channel_type was negotiated rather than the feature bit?

Reply:
Yea, the current text of this PR, however, only adds the only-use-alias-when-forwarding requirement for the "`channel_type` has `option_scid_alias` set", which is correct, I believe. There is nothing implied by the feature bit here.

Reply:
For private zero-conf channels, this line seems to be ignored (the zero-conf and option-scid-alias channel types seem to be exclusive)

Reply:
I'm honestly entirely unsure what we're discussing here at this point - the above comments all seem to be on the same page - you can't set scid_alias on a public channel, and for private channels you can set it allowing the limiting of relay to the alias.